### DISCUSSION TITLE
API Restriction

### DISCUSSION THREAD

**Discussion by adamauclair** [cloudbet/docs#21] [General]
Hi,
After a security verification my cloudbet account,  I was asked to place bet only through the API.
Unfortunately, when I turned back on my bot the API responses to my bets were  always "restricted".
I asked cloudbet support for some help, they answered my that they do not handle api queries and that I should raise a ticket at
https://github.com/cloudbet/docs
So here I am.
Thank you

**Comment by kgravenreuth** [comment]
Hi @adamauclair 

We have further reviewed your account and you should be able to place bets via our API now. Please be aware of all the Cloudbet Terms and Conditions going forward so that we can keep a fun and fruitful environment for the Cloudbet community. Specifically, please take note of the following clause from the [Cloudbet Terms and Conditions page](https://www.cloudbet.com/en/help/terms?section=binding-declarations&utm_source=github.com&utm_medium=community&utm_campaign=axis): 

_• When using the API be considerate. If the bet rejection rate on the betting API is too high (e.g. > 80% on your last 100 bets) your account will be flagged for abusive behaviour. You may get blocked for up to 7 days for integrity checks._

If you have any other questions regarding our API, please raise them here and we'll get them clarified ASAP.

**Comment by adamauclair** [comment]
Thanks you [@kgravenreuth](https://github.com/kgravenreuth) for the quick answer. It's working again :-)

I often have bets rejected due to STAKE_ABOVE_MAX even if I pay attention to stake under the "maxStake" value 
obtained when I get odds with a requests on sports-api.cloudbet.com/pub/v2/odds/events/ .
In such cases, I use the api answer "stake" of the rejected bet and place again a bet with this value. 
It's working but I've got a high rejection rate.
"maxStake" value seems wrong for me, what is its unit / currency ?

**Comment by kgravenreuth** [comment]
Glad to hear that it's working for you again @adamauclair 

> "maxStake" value seems wrong for me, what is its unit / currency ?

`maxStake` and `stake` values are always denominated in `EUR` currency. 

Please note that the `maxStake` value from the Cloudbet Feed API is indicative, while the `stake` returned in the Trading API for rejected bets is the value you should be using as the definitive value for retrying bets rejected for invalid stakes. We only flag accounts when they repeatedly send invalid bet attempts, while re-attempting such bets should be your normal workflow.

You can refer to the cases where we would send new `stake` values with rejected bets in our [Trading API docs](https://www.cloudbet.com/api/?urls.primaryName=Trading#/Trading/PlaceBet?utm_source=github.com&utm_medium=community&utm_campaign=axis):

```
STAKE_ABOVE_MAX - stake requested was above the current maximum stake on a selection. Please reference stake from response payload about the corrected value for retry
STAKE_BELOW_MIN - stake requested was below the current minimum stake on a selection. Please reference stake from response payload about the corrected value for retry
LIABILITY_LIMIT_EXCEEDED - your current liability limit on this event was exceeded. Please reference stake from response payload about the corrected value for retry
```

**Reply by adamauclair** [reply]
Thank you for your precisions. 
Unfortunately I was restricted again today. This time, I think I know why :
One bet was rejected due do "STAKE_ABOVE_MAX" and "stake" value in the api reply was "0". 
In such case, thinking it's an error, I asked my bot to try again with the minimal stake allowed (1e-7 BTC roughly 0.004 € today, not an agressive move to me). Most of the time, it works. But today for one particular bet, it was rejected again and my bot tries again many times until "RESTRICTED". I did not expect such low stake bet would be rejected, and did not program my bot accordingly. I will fix that.
If I understand well [Cloudbet Terms and Conditions page](https://www.cloudbet.com/en/help/terms?section=binding-declarations&utm_source=github.com&utm_medium=community&utm_campaign=axis), I will have to wait 1 week before turning my bot back on.  It's not easy to build a robust bot.
